In [1]:
%pip install git+https://github.com/awslabs/fast-differential-privacy

  Cloning https://github.com/awslabs/fast-differential-privacy to /tmp/pip-req-build-zxwqt0li
  Running command git clone --filter=blob:none --quiet https://github.com/awslabs/fast-differential-privacy /tmp/pip-req-build-zxwqt0li
  Resolved https://github.com/awslabs/fast-differential-privacy to commit 1202baa441c38c338415bc24e4cf12514c46bac0
  Preparing metadata (setup.py) ... done
  Created wheel for fastDP: filename=fastDP-1.1.0-py3-none-any.whl size=101664 sha256=73d35ca0c523b023b60ab4dadb7e58f68d3453816cff2c89b25fd45f21ebdf42
  Stored in directory: /tmp/pip-ephem-wheel-cache-ptq8p9sr/wheels/b3/2b/b5/62668a08336b29e28ca790358738474aa8c2c68359b494f454
Successfully built fastDP


In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
%cd /content/drive/Othercomputers/My Mac/CLIPcifar10/

/content/drive/Othercomputers/My Mac/CLIPcifar10


In [ ]:
!bash lptrial.sh

Files already downloaded and verified
Files already downloaded and verified
Total parameters 23528522. Trainable parameters 20490.
Number of trainable components:  2 ; Number of trainable layers:  1
>>>>>>>>>>>>>>>>> Applying  automatic  per-sample gradient clipping.
>>>>>>>>>>>>>>>>> Block heads for per-sample gradient clipping are defined as: ['_model.fc']
/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py:1359: UserWarning: Using a non-full backward hook when the forward contains multiple autograd Nodes is deprecated and will be removed in future versions. This hook will be missing some grad_input. Please use register_full_backward_hook to get the documented behavior.
  warnings.warn("Using a non-full backward hook when the forward contains multiple autograd Nodes "
Accuracy of the network on the 10000 test images: 37 %
Accuracy of the network on the 10000 test images: 49 %
Accuracy of the network on the 10000 test images: 52 %
Accuracy of the network on the 10000 te

### s0e50eps1.5b180lr0.001 SGD result

11 %
,11 %
,11 %
,11 %
,11 %
,12 %
,12 %
,12 %
,12 %
,12 %
,12 %
,12 %
,13 %
,13 %
,13 %
,13 %
,13 %
,13 %
,13 %
,13 %
,13 %
,14 %
,14 %
,14 %
,14 %
,14 %
,14 %
,14 %
,14 %
,14 %
,15 %

### s0e50eps1.5b180lr0.01 SGD result

11 %
,11 %
,12 %
,13 %
,14 %
,17 %
,19 %
,21 %
,23 %
,26 %
,28 %
,30 %
,31 %
,34 %
,36 %
,37 %
,38 %
,40 %
,41 %
,42 %
,44 %
,45 %
,47 %
,48 %
,49 %
,50 %
,50 %

### s0e50eps1.5b180lr0.1 SGD result

32 %
,45 %
,51 %
,56 %
,59 %
,59 %
,58 %
,55 %
,56 %
,57 %
,57 %

### s0e50eps1.5b180lr1 SGD result

37 %
,49 %
,52 %
,60 %
,59 %
,60 %
,60 %